### DISCUSSION TITLE
Regarding flame spread tracking

### DISCUSSION THREAD

**Discussion by ashutoshtdceiitm** [firemodels/fds#13200] [Q&A]
Hi, I need some help regarding how should I track the flame spread in case of wildland fire spread. I am using boundary fuel model. Is there a specific method followed to track the flame front? Shall I use the gas phase temperature or fix any temperature of the pyrolysis front? How to probe the pyrolysis front temperature? Thank you

**Comment by ashutoshtdceiitm** [comment]
I have seen a question mentioning to create an array of points and track the temperature beyond a certain value, but is there a way to track pyrolysis front temperature and not the gas-phase one?

**Comment by mcgratta** [comment]
You can use the same technique (`POINTS` on a `DEVC` line) to track solid phase quantities like `'WALL TEMPERATURE'`.

**Reply by ashutoshtdceiitm** [reply]
Hi Kevin, Thank you for providing a direction. Now, I have included both phases, solid as well gas to track the spread. 
Going ahead, it would be great if you can explain a bit more about fire depth and how the following can be used for this purpose. I found these in the validation cases for wildland fires, but feeling difficulty implementing as I need to understand the threshold setpoint as mentioned in Section 18.3.8 of the User Guide.
&DEVC ID='x', QUANTITY='TEMPERATURE', SPATIAL_STATISTIC='MAXLOC X', XB=-20,220,-1,0,0,1 /
&DEVC ID='D', QUANTITY='FIRE DEPTH', XB=-20,220,0,0,0.1,0.1 /

**Reply by mcgratta** [reply]
I do not think that `'FIRE DEPTH'` is going to be helpful. All that it does is calculate the "width" of the fire front, but it tells you nothing about its speed.

**Reply by ashutoshtdceiitm** [reply]
Thanks Kevin. I will explore this.

**Comment by TristanHehnen** [comment]
You may need to set up some of your own logic to track flame fronts. One way could be to track the `HRRPUV` in the gas cells directly above the ground to ensure combustion. You may need to account for leading and trailing edges of the flame itself to ensure that really the front is tracked. 
You could have a look at this article, even though it is not focused on wild land fire it could be a good starting point. There is also a data set with the Python script for download:
https://doi.org/10.1016/j.firesaf.2024.104116

**Reply by ashutoshtdceiitm** [reply]
Hi Tristan, Thanks for this. I am looking into it.
